src/signtool/IBM_Container.h

```diff
@@ -0,0 +1,253 @@
+                                                 // holder ecid_count * szeof(ecids)
+    SoftwareHdr();
+                                             
+    void PrintHeader() const;
+    void GetHeaderBytes( std::vector<uint8_t>& packet ) const;
+
+} __attribute__ ((packed));
+
+
+struct SoftwareSig
+{
+    uint8_t   m_swSigP[ECDSA521_SIG_SIZE];
+    uint8_t   m_swSigQ[ECDSA521_SIG_SIZE];
+    uint8_t   m_swSigR[ECDSA521_SIG_SIZE];
+
+    SoftwareSig();
+                                             
+    void PrintHeader() const;
+    void GetHeaderBytes( std::vector<uint8_t>& packet ) const;
+
+} __attribute__ ((packed));
+
+
+
+
+/* The Container Layout consists of the following 5 blocks 
+ *   ContainerHdr
+ *   PrefixHdr
+ *   PrefixData
+ *   SoftwareHdr
+ *   SoftwareSig
+ */
+class IBM_Container
+{
+public:
+    enum IBM_HdrFldType
+    {
+        e_FLD_PREFIX_HDR,
+        e_FLD_SOFTWARE_HDR
+    };
+
+    enum IBM_ContainerFld
+    {
+        // these are the fields in Container Header
+        e_CONTAINER_VERSION,
+        e_CONTAINER_SIZE,
+        e_TARGET_HRMOR,
+        e_STACK_POINTER,
+        e_HW_PUBLIC_KEY_A,
+        e_HW_PUBLIC_KEY_B,
+        e_HW_PUBLIC_KEY_C,
+   
+        // these are the fields in Prefix Header
+        e_PRE_HDR_VERSION,
+        e_PRE_HDR_HASH_ALGORITHM,
+        e_PRE_HDR_SIGNING_ALGORITHM,
+        e_PRE_HDR_CODE_START_OFFSET,
+        e_PRE_HDR_FLAGS,
+        e_PRE_HDR_PAYLOAD_SIZE,
+        e_PRE_HDR_PAYLOAD_HASH,
+
+        // these are the fields in Prefix Data
+        e_HW_SIGNATURE_A,
+        e_HW_SIGNATURE_B,
+        e_HW_SIGNATURE_C,
+        e_SW_PUBLIC_KEY_P,
+        e_SW_PUBLIC_KEY_Q,
+        e_SW_PUBLIC_KEY_R,
+
+        // these are the fields in Software Header
+        e_SW_HDR_VERSION,
+        e_SW_HDR_HASH_ALGORITHM,
+        e_SW_HDR_CODE_START_OFFSET,
+        e_SW_HDR_FLAGS,
+        e_SW_HDR_PAYLOAD_SIZE,
+        e_SW_HDR_PAYLOAD_HASH,
+
+        // these are the fields in Software Signature
+        e_SW_SIGNATURE_P,
+        e_SW_SIGNATURE_Q,
+        e_SW_SIGNATURE_R
+    };
+
+    // default C'tor
+    IBM_Container( std::string p_mode );
+
+    // Given a filename, read its contents, parse the data and constuct the container
+    IBM_Container( std::string p_mode,
+                   std::string p_containerFileName );
+
+    // Given a stream of bytes, parse the data and constuct the container
+    IBM_Container( uint8_t *p_rawData );
+
+    ~IBM_Container();
+
+    int  Validate();
+
+    void Print() const;
+
+    bool Save( const std::string p_fileName ); 
+
+    bool UpdateField( const std::string p_fldName, const std::string p_value ); 
+
+    void GetFieldNameList( std::vector<std::string>& p_fldNameList );
+
+    bool ComputeHash( std::string  p_hashHdrType,
+                      std::string  p_hashAlgo,
+                      std::string& p_digestStr );
+
+private:
+    // Disallow Copy Constructor and Assignment Operator
```

Comment:
You also need to deal with the move functions.  And the preferred way in C++11 is `=delete`, not making them private.  
Private doesn't entirely solve the problem because friends are still impacted and it causes linker-time errors instead of compile-time errors.